x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/Transaction/index.tsx

```diff
@@ -82,11 +82,22 @@ function MaybeViewTraceLink({
 
     // the user is viewing a zoomed in version of the trace. Link to the full trace
   } else {
+    const traceRoot = waterfall.traceRoot;
     return (
       <EuiFlexItem grow={false}>
-        <TransactionLink transaction={waterfall.traceRoot}>
-          <EuiButton iconType="apmTrace">{viewFullTraceButtonLabel}</EuiButton>
-        </TransactionLink>
+        {traceRoot && (
```

Comment:
Do you need to check for the existence of `traceRoot`? We already have a check above `if(!waterfall.traceRoot)`

Reply:
nope, thanks, didn't see that